src/Traits/ClientEndpointsTrait.php

```diff
@@ -1993,44 +2035,44 @@ public function update(array $params = [])
-	 *     requests_per_second: number, // The throttle to set on this request in sub-requests per second. -1 means no throttle.
-	 *     slices: number|string, // The number of slices this task should be divided into. Defaults to 1, meaning the task isn't sliced into subtasks. Can be set to `auto`.
-	 *     pretty: boolean, // Pretty format the returned JSON response. (DEFAULT: false)
-	 *     human: boolean, // Return human readable values for statistics. (DEFAULT: true)
-	 *     error_trace: boolean, // Include the stack trace of returned errors. (DEFAULT: false)
-	 *     source: string, // The URL-encoded request definition. Useful for libraries that do not accept a request body for non-POST requests.
-	 *     filter_path: list, // A comma-separated list of filters used to reduce the response.
-	 *     body: array, //  The search definition using the Query DSL
+	 *     index: string, // (REQUIRED) A comma-separated list of index names to search; use `_all` or empty string to perform the operation on all indices
+	 *     analyzer?: string, // The analyzer to use for the query string
+	 *     analyze_wildcard?: bool, // Specify whether wildcard and prefix queries should be analyzed (default: false)
+	 *     default_operator?: string, // The default operator for query string query (AND or OR)
+	 *     df?: string, // The field to use as default where no field prefix is given in the query string
+	 *     from?: float, // Starting offset (default: 0)
+	 *     ignore_unavailable?: bool, // Whether specified concrete indices should be ignored when unavailable (missing or closed)
+	 *     allow_no_indices?: bool, // Whether to ignore if a wildcard indices expression resolves into no concrete indices. (This includes `_all` string or when no indices have been specified)
+	 *     conflicts?: string, // What to do when the update by query hits version conflicts?
+	 *     expand_wildcards?: string, // Whether to expand wildcard expression to concrete indices that are open, closed or both.
+	 *     lenient?: bool, // Specify whether format-based query failures (such as providing text to a numeric field) should be ignored
+	 *     pipeline?: string, // Ingest pipeline to set on index requests made by this action. (default: none)
+	 *     preference?: string, // Specify the node or shard the operation should be performed on (default: random)
+	 *     q?: string, // Query in the Lucene query string syntax
+	 *     routing?: string, // A comma-separated list of specific routing values
+	 *     scroll?: int|string, // Specify how long a consistent view of the index should be maintained for scrolled search
+	 *     search_type?: string, // Search operation type
+	 *     search_timeout?: int|string, // Explicit timeout for each search request. Defaults to no timeout.
+	 *     max_docs?: float, // Maximum number of documents to process (default: all documents)
```

Comment:
Consider if `int|float` should be the type for `number` as some of these do not appear to make sens as fractions. If it can accept a string representation then you could set it as `numeric` as that is short for `int|float|numeric-string`

Reply:
For the Elasticsearch specification reported [here](https://github.com/elastic/elasticsearch/tree/v8.17.3/rest-api-spec#type-definition), a `number` is alias for double (moreover it's deprecated). I agree that makes no sense for `max_docs` that should be always an integer. I will double check with the Elasticsearch team. Thanks!

Reply:
Ok, in that case it sounds like the issue isn't related to this PR but more the upstream documentation 👍